code/modules/projectiles/ammunition.dm

```diff
@@ -165,3 +165,8 @@
 //Behavior for magazines
 /obj/item/ammo_box/magazine/proc/ammo_count()
 	return stored_ammo.len
+
+/obj/item/ammo_box/magazine/proc/empty_magazine()
+	for(var/obj/item/ammo in stored_ammo)
+		ammo.forceMove(get_turf(src))

```

Comment:
store ```get_turf(src)``` in a var before the loop, it'll always be the same inside the loop, so no point doing it n times instead of once.